### ISSUE TITLE
Stream has already been listened to after 2.0.1 -> 2.0.2 update

### ISSUE THREAD

**Issue by eye-dee** [pratikbaid3/flutter_client_sse#25]
In the app, I have one screen calling `SSEClient.subscribeToSSE`, consecutive screen calls different server endpoint, but it also uses the same library method and I am getting the following exception:

```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Bad state: Stream has already been listened to.
#0      _StreamController._subscribe (dart:async/stream_controller.dart:686:7)
#1      _ControllerStream._createSubscription (dart:async/stream_controller.dart:837:19)
#2      _StreamImpl.listen (dart:async/stream_impl.dart:497:9)
#3      new _ForwardingStreamSubscription (dart:async/stream_pipe.dart:114:10)
#4      _ForwardingStream._createSubscription (dart:async/stream_pipe.dart:86:16)
#5      _ForwardingStream.listen (dart:async/stream_pipe.dart:81:12)
#6      new _ForwardingStreamSubscription (dart:async/stream_pipe.dart:114:10)
#7      _ForwardingStream._createSubscription (dart:async/stream_pipe.dart:86:16)
#8      _ForwardingStream.listen (dart:async/stream_pipe.dart:81:12)
#9      new _ForwardingStreamSubscription (dart:async/stream_pipe.dart:114:10)
#10     _ForwardingStream._createSubscription (dart:async/stream_pipe.dart:86:16)
#11     _ForwardingStream.listen (dart:asyn<…>
```

**Comment by soufianebenyaala** [comment]
same here

**Comment by pratikbaid3** [comment]
Hi @eye-dee and @soufianebenyaala . I am working on a fix for this issue. Will raise a PR today.

**Comment by pratikbaid3** [comment]
https://github.com/pratikbaid3/flutter_client_sse/pull/32

@pedroculque @soufianebenyaala @eye-dee. I have created a fix for this. It will be great if you would try it out to confirm if it fixes your issue.

**Comment by pratikbaid3** [comment]
Issue has been fixed in 2.0.3
